src/prefect/_internal/concurrency/services.py

```diff
@@ -122,6 +134,11 @@ def _drain(self) -> concurrent.futures.Future:
         Internal implementation for `drain`. Returns a future for sync/async interfaces.
         """
         self._stop()
+        if self._done_event.is_set():
```

Comment:
When does this case occur? If we've already called `_drain` before?

Reply:
Yeah basically repeated calls to `drain` i.e. the `drain` call trigger by atexit after a manual drain